### DISCUSSION TITLE
Unable to create parts warning on creation and CH startup

### DISCUSSION THREAD

**Discussion by christo-ph** [ClickHouse/ClickHouse#64533] [Q&A] [answered]
Following the example of https://clickhouse.com/docs/en/optimize/sparse-primary-indexes#a-table-with-a-primary-key to create that table, I get the warning

```
2024.05.28 18:24:03.855160 [ 13485536 ] {ab77dd6b-0e0a-48d1-841d-eb95992a4e80} <Warning> default.hits_UserID_URL (9d0aab9b-1ffb-4b53-8b63-7235f3800190): Table can't create parts with adaptive granularity, but settings min_rows_for_wide_part = 0, min_bytes_for_wide_part = 10485760. Parts with non-adaptive granularity can be stored only in Wide (default) format. Settings 'min_rows_for_wide_part'and 'min_bytes_for_wide_part' will be ignored.
```

What implication does that have? Otherwise the example seems to work properly and I get also the performance advantage mentioned in the doc.

-- 
ClickHouse 24.6.1.2175 (revision: 54487, git hash: 554a8cd7c29996040f9d5dafa0950fa702ec42a9, build id: <unknown>)
on MacOS

**Comment by den-crane** [comment]
`index_granularity_bytes=0` disables adaptive granularity. 
Don't use this setting. I think it's in the documentation by mistake.

Compact parts are not possible without adaptive granularity. And this warning explains it.

`index_granularity_bytes=0` / `min_bytes_for_wide_part = 10485760 (default)` / `min_rows_for_wide_part = 0 (default)` -- impossible combination.

**Reply by den-crane** [reply, marked as answer]
> I think it's in the documentation by mistake. 

ah. I see. It's not a mistake. It's for demonstration purposes for the article. To force wide parts

>The table’s data is stored in [wide format]
